code/game/objects/items/twohanded.dm

```diff
@@ -962,3 +963,169 @@
+	qdel(spark_system)
+	spark_system = null
+	return ..()
+
+/obj/item/twohanded/vxtvulhammer/update_icon()
+	icon_state = "vxtvul_hammer[wielded]-[is_supercharged]"
+
+/obj/item/twohanded/vxtvulhammer/examine(mob/living/carbon/user)
+	. = ..()
+	if(is_supercharged == 1)
+		. += "<b>Electric sparks</b> are bursting from the blast pads!"
+
+/obj/item/twohanded/vxtvulhammer/unwield(mob/living/carbon/user)
+	..()
+	if(is_supercharged == 1) //So you can't one-hand the charged hit
+		to_chat(user, span_notice("Your hammer loses its power as you adjust your grip."))
+		user.visible_message(span_warning("The sparks from [user]'s hammer suddenly stop!"))
+		supercharge(user)
+	if(is_charging == 1) //So you can't one-hand while charging
+		to_chat(user, span_notice("You flip the switch off as you adjust your grip."))
+		user.visible_message(span_warning("[user] flicks the hammer off!"))
+		is_charging = 0
+
+/obj/item/twohanded/vxtvulhammer/attack(atom/target, mob/living/carbon/user) //This doesn't consider objects, only people
+	if (is_charging == 1) //So you can't attack while charging
+		to_chat(user, span_notice("You flip the switch off before your attack."))
+		user.visible_message(span_warning("[user] flicks the hammer off and raises it!"))
+		is_charging = 0
+	..()
+
+/obj/item/twohanded/vxtvulhammer/AltClick(mob/living/carbon/user)
+	charge_hammer(user)
+
+/obj/item/twohanded/vxtvulhammer/proc/supercharge(mob/living/carbon/user) //Proc to handle when it's charged for light + sprite
+	if(is_supercharged == 0)
+		is_supercharged = 1
+		set_light(2) //Glows when charged
+	else
+		is_supercharged = 0
+		set_light(0)
+	update_icon()
+
+/obj/item/twohanded/vxtvulhammer/proc/charge_hammer(mob/living/carbon/user)
+	if(wielded == FALSE)
+		to_chat(user, span_warning("The hammer must be wielded in two hands in order to charge it!"))
+		return
+	if(is_supercharged == 1)
+		to_chat(user, span_warning("The hammer is already supercharged!"))
+	else
+		is_charging = 1
+		to_chat(user, span_notice("You begin charging the weapon, concentration flowing into it..."))
+		user.visible_message(span_warning("[user] flicks the hammer on, tilting their head down as if in thought."))
+		spark_system.start() //Generates sparks when you charge
+		if(!do_mob(user, user, 60)) //Should be six seconds
+			if(is_charging == 0) //So no duplicate messages
+				return
+			to_chat(user, span_notice("You flip the switch off as you lose your focus."))
+			user.visible_message(span_warning("[user]'s concentration breaks!"))
+			is_charging = 0
+		if(is_charging == 0) //No charging for you if you cheat
+			return
+		to_chat(user, span_notice("You complete charging the weapon."))
+		user.visible_message(span_warning("[user] looks up as their hammer begins to crackle and hum!"))
+		playsound(loc, 'sound/magic/lightningshock.ogg', 60, 1) //Mainly electric crack
+		playsound(loc, 'sound/effects/magic.ogg', 40, 1) //Reverb undertone
+		supercharge(user)
+		is_charging = 0
```

Comment:
```suggestion
	if(wielded == FALSE)
		to_chat(user, span_warning("The hammer must be wielded in two hands in order to charge it!"))
		return
	if(supercharged)
		to_chat(user, span_warning("The hammer is already supercharged!"))
	else
		charging = TRUE
		to_chat(user, span_notice("You begin charging the weapon, concentration flowing into it..."))
		user.visible_message(span_warning("[user] flicks the hammer on, tilting their head down as if in thought."))
		spark_system.start() //Generates sparks when you charge
		if(!do_mob(user, user, 6 SECONDS))
			if(!charging) //So no duplicate messages
				return
			to_chat(user, span_notice("You flip the switch off as you lose your focus."))
			user.visible_message(span_warning("[user]'s concentration breaks!"))
			charging = FALSE
		if(!charging) //No charging for you if you cheat
			return //Has to double-check return because attacking or one-handing won't actually proc !do_mob, so the channel will seem to continue despite the message that pops out, but this actually ensures that it won't charge despite attacking or one-handing
		to_chat(user, span_notice("You complete charging the weapon."))
		user.visible_message(span_warning("[user] looks up as their hammer begins to crackle and hum!"))
		playsound(loc, 'sound/magic/lightningshock.ogg', 60, TRUE) //Mainly electric crack
		playsound(loc, 'sound/effects/magic.ogg', 40, TRUE) //Reverb undertone
		supercharge()
		charging = FALSE
```